inc/server/class-form-server.php

```diff
@@ -7,56 +7,135 @@
+ *
+ * @since 2.0.0
  */
 class Form_Server {
 
 	/**
 	 * The main instance var.
 	 *
 	 * @var Form_Server
+	 * @since 2.0.0
 	 */
 	public static $instance = null;
 
 	/**
 	 * Rest route namespace.
 	 *
 	 * @var Form_Server
+	 * @since 2.0.0
 	 */
 	public $namespace = 'otter/';
 
 	/**
 	 * Rest route version.
 	 *
 	 * @var Form_Server
+	 * @since 2.0.0
 	 */
 	public $version = 'v1';
 
+
 	/**
 	 * Initialize the class
+	 *
+	 * @since 2.0.0
 	 */
 	public function init() {
+		/**
+		 * Register the REST API endpoints.
+		 */
 		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+
+		/**
+		 * Add filter for validating the form data.
+		 */
+		add_filter( 'otter_form_validation', array( $this, 'check_form_conditions' ) );
+
+		/**
+		 * Add filter for captcha validation.
+		 */
+		add_filter( 'otter_form_captcha_validation', array( $this, 'check_form_captcha' ) );
+
+		/**
+		 * Register email providers that can be used to send emails or subscribe to a service.
+		 */
+		$defaultProvider = array(
+			'frontend' => array(
+				'submit' => array( $this, 'send_default_email' ),
+			),
+			'editor'   => array(
+				'listId'    => array( $this, 'get_integration_data' ),
+				'testEmail' => array( $this, 'send_test_email' ),
+			),
+		);
+
+		// Register 3rd party email providers.
+		$sendinblueProvider = array(
+			'frontend' => array(
+				'submit' => array( $this, 'subscribe_to_service' ),
+			),
+			'editor'   => array(
+				'listId'    => array( $this, 'get_integration_data' ),
+				'testEmail' => array( $this, 'test_subscription_service' ),
+			),
+		);
+		$mailchimpProvider  = array(
+			'frontend' => array(
+				'submit' => array( $this, 'subscribe_to_service' ),
+			),
+			'editor'   => array(
+				'listId'    => array( $this, 'get_integration_data' ),
+				'testEmail' => array( $this, 'test_subscription_service' ),
+			),
+		);
+
+		do_action(
+			'otter_register_form_providers',
+			array(
+				'default'    => $defaultProvider,
+				'sendinblue' => $sendinblueProvider,
+				'mailchimp'  => $mailchimpProvider,
+			)
+		);
+
+		add_action( 'otter_form_before_submit', array( $this, 'before_submit' ) );
+		add_action( 'otter_form_after_submit', array( $this, 'after_submit' ) );
+		add_filter( 'otter_form_email_build_body', array( $this, 'build_email_content' ) );
+		add_filter( 'otter_form_email_build_body_error', array( $this, 'build_email_error_content' ), 1, 2 );
 	}
 
 	/**
 	 * Register REST API route
+	 *
+	 * @since 2.0.0
 	 */
 	public function register_routes() {
 		$namespace = $this->namespace . $this->version;
 		register_rest_route(
 			$namespace,
-			'/forms',
+			'/form/frontend',
 			array(
 				array(
-					'methods'             => \WP_REST_Server::CREATABLE,
-					'callback'            => array( $this, 'submit_form' ),
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'frontend' ),
 					'permission_callback' => function () {
```

Comment:
Can we put some checks to make sure no one can simply ping this HTTP without a proper nonce. Currently if I just send dummy data, it might start triggering error emails. Having some check to make sure the user is triggering this request from the website, not thru Postman.